overlay/etc/init.d/net-online

```diff
@@ -0,0 +1,72 @@
+#!/sbin/openrc-run
+# Copyright (c) 2015 The OpenRC Authors.
+# See the Authors file at the top-level directory of this distribution and
+# https://github.com/OpenRC/openrc/blob/master/AUTHORS
+#
+# This file is part of OpenRC. It is subject to the license terms in
+# the LICENSE file found in the top-level directory of this
+# distribution and at https://github.com/OpenRC/openrc/blob/master/LICENSE
+# This file may not be copied, modified, propagated, or distributed
+# except according to the terms contained in the LICENSE file.
+
+description="Delays until the network is online or a specific timeout"
+
+depend()
+{
+	after modules net
+	need sysfs
+	provide network-online
+	keyword -docker -jail -lxc -openvz -prefix -systemd-nspawn -uml -vserver
+}
+
+get_interfaces()
+{
+	local ifname iftype
+	for ifname in /sys/class/net/*; do
+		[ -h "${ifname}" ] || continue
+		[ -e "/sys/devices/virtual/net/${ifname##*/}" ] && continue
+		read iftype < ${ifname}/type
+		[ "$iftype" = "1" ] && printf "%s " ${ifname##*/}
+	done
+}
+
+start ()
+{
+	local carriers configured dev gateway ifcount infinite
+	local carrier operstate rc
+
+	ebegin "Checking to see if the network is online"
+	rc=0
+	interfaces=${interfaces:-$(get_interfaces)}
+	timeout=${timeout:-120}
+	[ $timeout -eq 0 ] && infinite=true || infinite=false
+	while $infinite || [ $timeout -gt 0 ]; do
+	       carriers=0
+	       configured=0
+	       ifcount=0
+		for dev in ${interfaces}; do
+			[ -e /sys/class/net/$dev ] || continue
+		: $((ifcount += 1))
+		read carrier < /sys/class/net/$dev/carrier >/dev/null 2>&1 || carrier=
+		[ "$carrier" = 1 ] && : $((carriers += 1))
+		read operstate < /sys/class/net/$dev/operstate >/dev/null 2>&1 || operstate=
+		[ "$operstate" = up ] && : $((configured += 1))
+	       done
+	       [ $configured -ge 1 ] && [ $carriers -ge 1 ] && break
```

Comment:
would probably do ${SOMEVAR:-ifcount} so we can configure the number of required devices to be online.  And we can default to 1.